cmd/lock-rpc-server.go

```diff
@@ -227,132 +272,41 @@ type nameLockRequesterInfoPair struct {
 
 		var active bool
 
 		// Call back to original server verify whether the lock is still active (based on name & uid)
-		if err := c.Call("Dsync.Active", &LockArgs{Name: nlrip.name, UID: nlrip.lri.uid}, &active); err != nil {
-			// We failed to connect back to the server that originated the lock, this can either be due to
-			// - server at client down
-			// - some network error (and server is up normally)
-			//
-			// We will ignore the error, and we will retry later to get resolve on this lock
-			c.Close()
-		} else {
-			c.Close()
-
-			if !active { // The lock is no longer active at server that originated the lock
-				// so remove the lock from the map
-				l.mutex.Lock()
-				// Check if entry is still in map (could have been removed altogether by 'concurrent' (R)Unlock of last entry)
-				if lri, ok := l.lockMap[nlrip.name]; ok {
-					if !l.removeEntry(nlrip.name, nlrip.lri.uid, &lri) {
-						// Remove failed, in case it is a:
-						if nlrip.lri.writer {
-							// Writer: this should never happen as the whole (mapped) entry should have been deleted
-							log.Errorln("Lock maintenance failed to remove entry for write lock (should never happen)", nlrip.name, nlrip.lri, lri)
-						} else {
-							// Reader: this can happen if multiple read locks were active and the one we are looking for
-							// has been released concurrently (so it is fine)
-						}
-					} else {
-						// remove went okay, all is fine
-					}
-				}
-				l.mutex.Unlock()
-			}
-		}
-	}
-}
-
-// Initialize distributed lock.
-func initDistributedNSLock(mux *router.Router, serverConfig serverCmdConfig) {
-	lockServers := newLockServers(serverConfig)
-	registerStorageLockers(mux, lockServers)
-}
-
-// Create one lock server for every local storage rpc server.
-func newLockServers(serverConfig serverCmdConfig) (lockServers []*lockServer) {
-	// Initialize posix storage API.
-	exports := serverConfig.disks
-	ignoredExports := serverConfig.ignoredDisks
-
-	// Save ignored disks in a map
-	skipDisks := make(map[string]bool)
-	for _, ignoredExport := range ignoredExports {
-		skipDisks[ignoredExport] = true
-	}
-	for _, export := range exports {
-		if skipDisks[export] {
-			continue
-		}
-		if isLocalStorage(export) {
-			if idx := strings.LastIndex(export, ":"); idx != -1 {
-				export = export[idx+1:]
-			}
-
-			// Create handler for lock RPCs
-			locker := &lockServer{
-				rpcPath:   export,
-				mutex:     sync.Mutex{},
-				lockMap:   make(map[string][]lockRequesterInfo),
-				timestamp: time.Now().UTC(),
-			}
-
-			// Start loop for stale lock maintenance
-			go func() {
-				// Start with random sleep time, so as to avoid "synchronous checks" between servers
-				time.Sleep(time.Duration(rand.Float64() * float64(lockMaintenanceLoop)))
-				for {
-					time.Sleep(lockMaintenanceLoop)
-					locker.lockMaintenance(lockCheckValidityInterval)
-				}
-			}()
-
-			lockServers = append(lockServers, locker)
+		err := c.Call("Dsync.Active", &LockArgs{
```

Comment:
If we negate the function of 'Dsync.Active', thus instead do a 'Dsync.Expired', then we can do as follows (and get rid of `err` altogether)

```
var expired bool
// Purposely ignore error
c.Call("Dsync.Expired", &LockArgs{.....}, &expired)
c.Close()
if expired { .....
}
```